internal/hcl/remote_variables_loader.go

```diff
@@ -0,0 +1,237 @@
+		Attributes struct {
+			Name   string `json:"name"`
+			Global bool   `json:"global"`
+		} `json:"attributes"`
+		Relationships struct {
+			Vars struct {
+				Data []struct {
+					ID string
+				} `json:"data"`
+			} `json:"vars"`
+		} `json:"relationships"`
+	}
+
+	type tfcVar struct {
+		Key       string `json:"key"`
+		Value     string `json:"value"`
+		Sensitive bool   `json:"sensitive"`
+		Category  string `json:"category"`
+	}
+
+	var varsetsResponse struct {
+		Data     []tfcVarset `json:"data"`
+		Included []struct {
+			ID         string `json:"id"`
+			Type       string `json:"type"`
+			Attributes tfcVar `json:"attributes"`
+		} `json:"included"`
+		Meta struct {
+			Pagination struct {
+				NextPage int `json:"next-page"`
+			} `json:"pagination"`
+		} `json:"meta"`
+	}
+	pageNumber := 1
+
+	varsets := []tfcVarset{}
+	varsMap := map[string]tfcVar{}
+
+	for {
+		endpoint = fmt.Sprintf("/api/v2/workspaces/%s/varsets?include=vars&page[number]=%d&page[size]=50", workspaceID, pageNumber)
+		body, err = client.Get(endpoint)
+		if err != nil {
+			return vars
+		}
+		if json.Unmarshal(body, &varsetsResponse) != nil {
+			return vars
+		}
+
+		varsets = append(varsets, varsetsResponse.Data...)
+		varsetVars := varsetsResponse.Included
+
+		for _, v := range varsetVars {
+			if v.Type == "vars" {
+				varsMap[v.ID] = v.Attributes
+			}
+		}
+
+		if varsetsResponse.Meta.Pagination.NextPage > pageNumber {
+			pageNumber = varsetsResponse.Meta.Pagination.NextPage
+		} else {
+			break
+		}
+	}
+
+	// Sort varsets alphabetically, global varsets are lower in priority and can
+	// be overridden by workspace's varsets.
+	sort.Slice(varsets, func(i, j int) bool {
+		if varsets[i].Attributes.Global && !varsets[j].Attributes.Global {
+			return true
+		}
+		if !varsets[i].Attributes.Global && varsets[j].Attributes.Global {
+			return false
+		}
+
+		return varsets[i].Attributes.Name > varsets[j].Attributes.Name
+	})
+
+	for _, varset := range varsets {
+		for _, v := range varset.Relationships.Vars.Data {
+			vv, ok := varsMap[v.ID]
+			if ok {
+				if vv.Sensitive || vv.Category != "terraform" || vv.Value == "" {
+					continue
+				}
+
+				vars[vv.Key] = cty.StringVal(vv.Value)
+			}
+		}
+	}
+
+	return vars
+}
+
+func (r *RemoteVariablesLoader) getOrganizationWorkspace(blocks Blocks) (string, string, error) {
+	organization := ""
+	workspace := ""
+
+	for _, block := range blocks {
+		if block.Type() != "terraform" {
+			continue
+		}
+
+		for _, c := range block.childBlocks {
+			if c.Type() != "cloud" {
+				continue
+			}
+
+			organization = r.getAttribute(c, "organization")
+
+			for _, cc := range c.childBlocks {
+				if cc.Type() != "workspaces" {
+					continue
+				}
+
+				workspace = r.getAttribute(cc, "name")
+
+				break
+			}
+
+			break
+		}
+
+		break
+	}
+
+	if organization == "" || workspace == "" {
+		return "", "", errors.Errorf("unable to detect organization or workspace")
+	}
+
+	return organization, workspace, nil
```

Comment:
```suggestion
	return "", "", errors.Errorf("unable to detect organization or workspace")
```

as change above returns early